package/MDAnalysis/core/selection.py

```diff
@@ -349,24 +369,27 @@ def __init__(self, parser, tokens):
         self.sel = parser.parse_expression(self.precedence)
 
     def _apply_KDTree(self, group):
-        """Selection using KDTree but periodic = True not supported.
-        (KDTree routine is ca 15% slower than the distance matrix one)
+        """Selection using KDTree
         """
         sel = self.sel.apply(group)
-        ref = sel.center_of_geometry()
-
-        kdtree = KDTree(dim=3, bucket_size=10)
+        box = self.validate_dimensions(group.dimensions)
+        ref = sel.center_of_geometry(pbc=self.periodic)
+        if box is None:
+            kdtree = KDTree(dim=3, bucket_size=10)
+        else:
+            kdtree = PeriodicKDTree(box, bucket_size=10)
         kdtree.set_coords(group.positions)
         kdtree.search(ref, self.cutoff)
         found_indices = kdtree.get_indices()
-
         return group[found_indices].unique
 
     def _apply_distmat(self, group):
         sel = self.sel.apply(group)
-        ref = sel.center_of_geometry().reshape(1, 3).astype(np.float32)
-
-        box = group.dimensions if self.periodic else None
+        periodic = False if self.validate_dimensions(group.dimensions) is None\
```

Comment:
shorter
```python
periodic = self.validate_dimensions(group.dimensions) is not None
```

Reply:
But even better, rearrange your code to avoid calling the same method twice:
```python
box = self.validate_dimensions(group.dimensions)
periodic = box is not None
ref = sel.center_of_geometry(pbc=periodic).reshape(1, 3).astype(np.float32)
```